common/paramsbuilder/params.go

```diff
@@ -13,7 +13,18 @@ type ParamAssurance interface {
 // Apply will apply options to construct a ready to go set of parameters.
 // This is a generalized constructor of parameters used to initialize any connector.
 // To qualify as a parameter one must have data validation laid out.
-func Apply[P ParamAssurance](params P, opts []func(params *P)) (*P, error) {
+//
+// It is possible to give default options which will be applied first.
+// Any other options supplied by the user may override defaults,
+// as they are applied last and therefore take higher precedence.
+func Apply[P ParamAssurance](params P,
+	opts []func(params *P),
+	defaultOpts ...func(params *P),
```

Comment:
Given connectors can be configured via options the only way to specify defaults a connector should have is to directly pass them. As option function sets values to parameter the last caller wins.